### ISSUE TITLE
LDAP: updating a user's role

### ISSUE THREAD

**Issue by SHSauler** [MISP/MISP#2368]
**TL/DR:** LDAP role assignment does not get updated.

### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       |  Ubuntu 16.04 LTS
| OS version (client)       | Yucky Windows
| PHP version               | PHP 7.0.18-0ubuntu0.16.04.1 (cli) ( NTS )
| MISP version / git hash   | v2.4.77
| Browser                   | Chrome

### Expected behavior
For an existing user in MISP, changing the LDAP group membership should change the role in MISP as well.

### Actual behavior
Even after removing cookie, the new role is not replicated in MISP. Only after removing the existing user and logging in again, the new role is assigned to the newly created user

### Steps to reproduce the behavior
* Assign LDAP role to new user
* New user logs in
* Change LDAP role of user
* User logs in again with old role assignment

Issue probably in [ApacheAuthenticate.php](https://github.com/MISP/MISP/blob/2922516e8412f387f1a14aac7a607274f8d85d89/app/Controller/Component/Auth/ApacheAuthenticate.php)?

**Comment by stinnux** [comment]
It's probably caused by #2251 . I will have a look into it tomorrow

**Comment by bmagistro** [comment]
Has there been any updates on this?  Recently tried to change a users role in LDAP but noticed it wasn't updated in MISP.

We are on CentOS 7.5 and MISP 2.4.95.

**Comment by 1earch** [comment]
Would PR #4389 help you?

```php
'updateUser' => true, // Optional : Will update user on LDAP login to update user fields (e.g. role)
```

**Comment by bmagistro** [comment]
@1earch I believe that PR came from someone else in our office and believe it does address the issue we were observing.
